src/Stratis.Bitcoin.Features.Dns/IFullNodeBuilderExtensions.cs

```diff
@@ -31,6 +32,9 @@ public static IFullNodeBuilder UseDns(this IFullNodeBuilder fullNodeBuilder)
                     services.AddSingleton<DnsSettings>();
                     services.AddSingleton<IUdpClient, DnsSeedUdpClient>();
                     services.AddSingleton<IWhitelistManager, WhitelistManager>();
+
+                    // Transient, means that every time this service is requested, a new instance is created.
+                    services.AddTransient<UnreliablePeerBehavior, UnreliablePeerBehavior>();
```

Comment:
hmm I wish, but we don't create behaviours from the DI so Transient wont matter here, so in that case just use a Singleton.

At some point we can refactor behaviours to use a factory with the DI then this will make sense.